tests/test_browser.py

```diff
@@ -4360,3 +4360,59 @@ def test_browser_modularize_no_current_script(self):
+
+  def test_emscripten_set_timeout(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_set_timeout.c'), '0', args=args)
+
+  def test_emscripten_set_timeout_loop(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_set_timeout_loop.c'), '0', args=args)
+
+  def test_emscripten_set_immediate(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_set_immediate.c'), '0', args=args)
+
+  def test_emscripten_set_immediate_loop(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_set_immediate_loop.c'), '0', args=args)
+
+  def test_emscripten_set_interval(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_set_interval.c'), '0', args=args)
+
+  def test_emscripten_date_now(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_date_now.c'), '0', args=args)
+
+  def test_emscripten_performance_now(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_performance_now.c'), '0', args=args)
+
+  def test_emscripten_console_log(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_console_log.c'), '0', args=args + ['--pre-js', path_from_root('tests', 'emscripten_console_log_pre.js')])
+
+  def test_emscripten_console_warn(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_console_warn.c'), '0', args=args + ['--pre-js', path_from_root('tests', 'emscripten_console_warn_pre.js')])
+
+  def test_emscripten_console_error(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_console_error.c'), '0', args=args + ['--pre-js', path_from_root('tests', 'emscripten_console_error_pre.js')])
+
+  def test_emscripten_throw_number(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_throw_number.c'), '0', args=args + ['--pre-js', path_from_root('tests', 'emscripten_throw_number_pre.js')])
+
+  def test_emscripten_throw_string(self):
+    for args in [[], ['-s', 'USE_PTHREADS=1', '-s', 'PROXY_TO_PTHREAD=1']]:
+      self.btest(path_from_root('tests', 'emscripten_throw_string.c'), '0', args=args + ['--pre-js', path_from_root('tests', 'emscripten_throw_string_pre.js')])
```

Comment:
I appreciate the modularity, but the test infrastructure makes each `btest` take a certain minimum time. perhaps some of these could naturally be combined together? if it's not natural though then nevermind. (we do need to optimize our browser testing harness at some point...)

Reply:
Reduced the number of tests by combining some, and moving other tests to only be tested either in threaded mode or non-threaded mode, with the assumption that they would also cover the other.